sqlglot/dialects/tsql.py

```diff
@@ -719,9 +719,30 @@ def _parse_user_defined_function(
             ):
                 return this
 
-            expressions = self._parse_csv(self._parse_function_parameter)
+            if not self._match(TokenType.WITH, advance=False):
+                expressions = self._parse_csv(self._parse_function_parameter)
+            else:
+                expressions = None
+
             return self.expression(exp.UserDefinedFunction, this=this, expressions=expressions)
 
+        def _parse_column_constraint(self) -> t.Optional[exp.Expression]:
+            if self._match(TokenType.CONSTRAINT):
+                this = self._parse_id_var()
+            else:
+                this = None
+
+            if not self._match(TokenType.WITH, advance=False) and self._match_texts(
+                self.CONSTRAINT_PARSERS
```

Comment:
This change looks more intrusive than necessary imho. I think the safer approach would be to check if `WITH` is followed by a token that signifies a procedure property instead of cutting this whole branch/logic off.

Reply:
As long as none of the "follow" keywords clash with existing constraint keywords, it should be fine to put this in the base parser and control the logic using a constant, e.g. something like:

```python
# Set this in T-SQL, and use an empty set for the base class, unless there's no collision with existing "WITH" constraints
PROCEDURE_WITH_PROPERTY_FOLLOW_KEYWORDS = {
    "ENCRYPTION",
    "EXECUTE",
    ...
}

def _parse_column_constraint(self) -> t.Optional[exp.Expression]:
    this = self._match(TokenType.CONSTRAINT) and self._parse_id_var()

    if (
        not self._match(TokenType.WITH)
        or not self._next
        or self._next.token_type not in self.PROCEDURE_WITH_PROPERTY_FOLLOW_KEYWORDS
    ):
        return self.expression(
            exp.ColumnConstraint,
            this=this,
            kind=self.CONSTRAINT_PARSERS[self._prev.text.upper()](self),
        )

    self._advance(self._index - 1)  # Unconsume the WITH token
    return this
```

Reply:
I decided to define PROCEDURE_OPTIONS in the Parser class as an empty OPTIONS_TYPE, and assign it the corresponding values in the TSQL class